python/cudf/cudf/pylibcudf_tests/test_column_factories.py

```diff
@@ -0,0 +1,282 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import pyarrow as pa
+import pytest
+from utils import DEFAULT_STRUCT_TESTING_TYPE, assert_column_eq
+
+from cudf._lib import pylibcudf as plc
+
+size = 3
+
+NUMERIC_TYPES = [
+    pa.uint8(),
+    pa.uint16(),
+    pa.uint32(),
+    pa.uint64(),
+    pa.int8(),
+    pa.int16(),
+    pa.int32(),
+    pa.int64(),
+    pa.float32(),
+    pa.float64(),
+    pa.bool_(),
+]
+
+TIMESTAMP_TYPES = [
+    pa.timestamp("s"),
+    pa.timestamp("ms"),
+    pa.timestamp("us"),
+    pa.timestamp("ns"),
+]
+
+DURATION_TYPES = [
+    pa.duration("s"),
+    pa.duration("ms"),
+    pa.duration("us"),
+    pa.duration("ns"),
+]
+
+DECIMAL_TYPES = [pa.decimal128(38, 2)]
+
+STRING_TYPES = [pa.string()]
+STRUCT_TYPES = [DEFAULT_STRUCT_TESTING_TYPE]
+LIST_TYPES = [pa.list_(pa.int64())]
+
+ALL_TYPES = (
+    NUMERIC_TYPES
+    + TIMESTAMP_TYPES
+    + DURATION_TYPES
+    + STRING_TYPES
+    + DECIMAL_TYPES
+    + STRUCT_TYPES
+    + LIST_TYPES
+)
+
+
+def pa_type_to_plc_type(pa_type):
+    # TODO: should be a cleaner way
+    return plc.interop.from_arrow(pa.array([], type=pa_type)).type()
+
+
+@pytest.fixture(
+    scope="module",
+    params=NUMERIC_TYPES,
+    ids=[
```

Comment:
suggestion: `ids=repr` gives an OK representation from the pyarrow type if you don't want to maintain a list of separate names